src/legacy/store/GraphQLStoreQueryResolver.js

```diff
@@ -246,8 +243,8 @@ class GraphQLStoreSingleQueryResolver {
       if (subscribedIDs) {
         // always subscribe to the root ID
         subscribedIDs[nextID] = true;
-        this._subscription = GraphQLStoreChangeEmitter.addListenerForIDs(
-          Object.keys(subscribedIDs),
+        this._subscription = this._storeData.getChangeEmitter().addListenerForIDs(
+          Object.keys(subscribedIDs).map(id => this._getCanonicalID(id)),
```

Comment:
That said, I think it might be a good idea to nix as many of these `this._storeData.get*()` calls as possible, and try to expose just the bare minimum API from `RelayStoreData`.

So this would just become `this._storeData.addListenerForIDs`, and `RelayStoreData` would take care of canonicalization and ushering the request to the change emitter.

Thoughts?